.evergreen/shared.yml

```
@@ -126,142 +138,173 @@ functions:
+          else
+            CYPRESS_CI=true CYPRESS_SKIP_VERIFY=true yarn cy:run --reporter junit
+          fi
       env:
         STAGING_RUNTIME_ENVIRONMENTS_BASE_URL: ${staging_runtime_environments_base_url}
         STAGING_RUNTIME_ENVIRONMENTS_API_KEY: ${staging_runtime_environments_api_key}
-      script: |
-        ${PREPARE_SHELL}
-        # Allow spec filtering for an intentional patch.
-        if [[ "${cypress_spec}" != "" ]]; then
-          CYPRESS_CI=true CYPRESS_SKIP_VERIFY=true yarn cy:run --reporter junit --spec "${cypress_spec}"
-        else
-          CYPRESS_CI=true CYPRESS_SKIP_VERIFY=true yarn cy:run --reporter junit
-        fi
 
   yarn-eslint:
-    command: shell.exec
+    command: subprocess.exec
     params:
       working_dir: ui/${app_dir}
-      shell: bash
-      script: |
-        ${PREPARE_SHELL}
-        yarn eslint:strict
+      binary: bash
+      args:
+        - -c
+        - |
+          ${PREPARE_SHELL}
+          yarn eslint:strict
 
   yarn-install:
-    command: shell.exec
+    command: subprocess.exec
     params:
       working_dir: ui/${app_dir}
-      shell: bash
-      script: |
-        ${PREPARE_SHELL}
-        yarn install
+      binary: bash
+      args:
+        - -c
+        - |
+          ${PREPARE_SHELL}
+          yarn install
 
   yarn-preview:
-    command: shell.exec
+    command: subprocess.exec
     params:
       working_dir: ui/${app_dir}
       background: true
-      shell: bash
-      script: |
-        ${PREPARE_SHELL}
-        yarn preview --no-request-logging
+      binary: bash
+      args:
+        - -c
+        - |
+          ${PREPARE_SHELL}
+          yarn preview --no-request-logging
 
   yarn-serve:
-    command: shell.exec
+    command: subprocess.exec
     type: setup
     params:
       working_dir: ui/${app_dir}
       background: true
-      script: |
-        ${PREPARE_SHELL}
-        yarn serve
+      binary: bash
+      args:
+        - -c
+        - |
+          ${PREPARE_SHELL}
+          yarn serve
 
   yarn-snapshot:
-    command: shell.exec
+    command: subprocess.exec
     params:
       working_dir: ui/${app_dir}
-      script: |
-        ${PREPARE_SHELL}
-        yarn snapshot
+      binary: bash
+      args:
+        - -c
+        - |
+          ${PREPARE_SHELL}
+          yarn snapshot
 
   yarn-storybook-build:
-    command: shell.exec
+    command: subprocess.exec
     params:
       working_dir: ui/${app_dir}
-      script: |
-        ${PREPARE_SHELL}
-        yarn storybook:build
+      binary: bash
+      args:
+        - -c
+        - |
+          ${PREPARE_SHELL}
+          yarn storybook:build
 
   yarn-test:
-    command: shell.exec
+    command: subprocess.exec
     params:
       working_dir: ui/${app_dir}
-      script: |
-        ${PREPARE_SHELL}
-        yarn test
+      binary: bash
+      args:
+        - -c
+        - |
+          ${PREPARE_SHELL}
+          yarn test
```

I think you can do a shorter one-liner for these repetitive ones:
```suggestion
　　　　args: ["-c", "source ${workdir}/.nvm/nvm.sh && yarn test"]
```